faiss_vector_posting.go

```diff
@@ -508,45 +510,49 @@ func (sb *SegmentBase) InterpretVectorIndex(field string, requiresFiltering bool
 							}
 						}
 
-						selector, err = faiss.NewIDSelectorBatch(vectorIDsToInclude)
+						if len(vectorIDsToInclude) > 0 {
+							selector, err = faiss.NewIDSelectorBatch(vectorIDsToInclude)
+						}
 					}
 					if err != nil {
 						return nil, err
 					}
 
-					// Ordering the retrieved centroid IDs by increasing order
-					// of distance i.e. decreasing order of proximity to query vector.
-					closestCentroidIDs, centroidDistances, _ :=
-						vecIndex.ObtainClustersWithDistancesFromIVFIndex(qVector,
-							eligibleCentroidIDs)
-
-					// Getting the nprobe value set at index time.
-					nprobe := vecIndex.GetNProbe()
-
-					eligibleDocsTillNow := int64(0)
-					minEligibleCentroids := 0
-					for i, centroidID := range closestCentroidIDs {
-						eligibleDocsTillNow += int64(centroidVecIDMap[centroidID].GetCardinality())
-						if eligibleDocsTillNow >= k && i >= int(nprobe-1) {
-							// Continue till at least 'K' cumulative vectors are
-							// collected or 'nprobe' clusters are examined, whichever
-							// comes later.
+					if len(vectorIDsToInclude) > 0 {
+						// Ordering the retrieved centroid IDs by increasing order
+						// of distance i.e. decreasing order of proximity to query vector.
+						closestCentroidIDs, centroidDistances, _ :=
+							vecIndex.ObtainClustersWithDistancesFromIVFIndex(qVector,
+								eligibleCentroidIDs)
+
+						// Getting the nprobe value set at index time.
+						nprobe := vecIndex.GetNProbe()
+
+						eligibleDocsTillNow := int64(0)
+						minEligibleCentroids := 0
+						for i, centroidID := range closestCentroidIDs {
+							eligibleDocsTillNow += int64(centroidVecIDMap[centroidID].GetCardinality())
+							if eligibleDocsTillNow >= k && i >= int(nprobe-1) {
+								// Continue till at least 'K' cumulative vectors are
+								// collected or 'nprobe' clusters are examined, whichever
+								// comes later.
+								minEligibleCentroids = i + 1
+								break
+							}
 							minEligibleCentroids = i + 1
-							break
 						}
-						minEligibleCentroids = i + 1
-					}
 
-					// Search the clusters specified by 'closestCentroidIDs' for
-					// vectors whose IDs are present in 'vectorIDsToInclude'
-					scores, ids, err := vecIndex.SearchClustersFromIVFIndex(
-						selector, len(vectorIDsToInclude), closestCentroidIDs,
-						minEligibleCentroids, k, qVector, centroidDistances, params)
-					if err != nil {
-						return nil, err
-					}
+						// Search the clusters specified by 'closestCentroidIDs' for
+						// vectors whose IDs are present in 'vectorIDsToInclude'
+						scores, ids, err := vecIndex.SearchClustersFromIVFIndex(
+							selector, len(vectorIDsToInclude), closestCentroidIDs,
+							minEligibleCentroids, k, qVector, centroidDistances, params)
+						if err != nil {
+							return nil, err
+						}
 
-					addIDsToPostingsList(rv, ids, scores)
+						addIDsToPostingsList(rv, ids, scores)
+					}
 					return rv, nil
```

Comment:
coding practice/maintenance wise: i think the usual practice in situations such

```
var value 
if cond {
  // changes to value specific to cond
  return value
}
return value
```
its better to avoid to the return statement present in the condition block since it anyways falls to the return value outside block.